README.md

# NetSUS Downloads

Installer:
[https://github.com/jamf/NetSUS/releases/download/5.0.2/NetSUSLPInstaller_5.0.2.run](https://github.com/jamf/NetSUS/releases/download/5.0.2/NetSUSLPInstaller_5.0.2.run)

OVA:
[https://github.com/jamf/NetSUS/releases/download/5.0.2/NetSUSLP_5.0.2.ova](https://github.com/jamf/NetSUS/releases/download/5.0.2/NetSUSLP_5.0.2.ova)

# Deprecation Notice

NetBoot and Software Update services have been deprecated by Apple, and as such this project will no longer be receiving further updates.

# What is NetSUS?

The NetSUSLP allows you to host an internal software update server (SUS), a NetBoot server, file shares, and a LDAP Proxy server **all on the same Linux system**. For a list of supported Linux distributions see [Requirements](#requirements).

<p align="center"><img src="docs/images/attachments/dashboard.png" height="400"></p>

* **Web Application** - The NetSUSLP includes a web application that can be used to easily manage your NetBoot, Software Update Servers and File Shares as well as your LDAP Proxy. The dashboard page is shown above.

* **File Sharing** - Use the NetSUSLP as a file share distribution point for Jamf Pro. You can share files using SMB, AFP, and HTTP.

* **Software Update Server** - Unlike a standard SUS, the SUS hosted by the NetSUSLP allows you to control which software updates should be installed on each computer in your organization.

* **NetBoot Server** - The NetSUSLP allows you to host a NetBoot image. You can boot computers to a NetBoot image in place of a recovery partition or external drive when imaging.

* **LDAP Proxy** - Use the NetSUSLP as a lightweight proxy that acts as a middleware layer between LDAP clients and LDAP directory servers.

## Documentation

For a getting started guide and step-by-step walkthroughs check out the **[documentation for the current release](docs/README.md)**

## <a name="requirements"></a>Requirements

#### Supported Linux distributions:

* Ubuntu 14.04 LTS Server
* Ubuntu 16.04 LTS Server (Recommended)
* Ubuntu 18.04 LTS Server
* Red Hat Enterprise Linux (RHEL) 6.4 or later
* CentOS 6.4 or later

#### To install the NetSUSLP using an installer, you need:

* The NetSUSLP Installer (.run), available from the [Releases](https://github.com/jamf/NetSUS/releases) page.
* 500 GB of disk space available 
* 1 GB of RAM

#### To set up the NetSUSLP as an appliance, you need:

* The OVA file for the NetSUSLP, available from the [Releases](https://github.com/jamf/NetSUS/releases) page.
* Virtualization software that supports Open Virtualization Format 
* 500 GB of disk space available
* 2 GB of RAM

#### If you are running a Kinobi Patch Server:

* Kinobi 1.0 is incompatible with NetSUS 5.0.2. If you are running Kinobi 1.0, you will need to install Kinobi 1.1 or later (1.3.2 is recommended) after upgrading to NetSUS 5.0.2, available from:
[https://github.com/mondada/kinobi/releases/download/1.3.2/KinobiInstaller_1.3.2.run](https://github.com/mondada/kinobi/releases/download/1.3.2/KinobiInstaller_1.3.2.run)

**Only Intel-based Macs can use a NetBoot server hosted by the NetSUSLP.**


CHANGELOG.md
# NetSUS Changelog

## 5.0.2

* Deprecation Notice - NetSUS is deprecated
* Updated reposado to the latest version
* Merged Pull Request #156, thanks [@wegotoeleven](https://github.com/wegotoeleven)
* Merged Pull Request #154, thanks [@sunny0day](https://github.com/sunny0day)
* Merged Pull Request #148, thanks [@Kaikon](https://github.com/Kaikon)
* Merged Pull Request #142, thanks [@james-tipler](https://github.com/james-tipler)

## 5.0.1

* Fixed a bug with SUS URL validation, when using SSL
* Fixed an issue with LDAP proxy install in some RHEL-based installations
* Fixed a bug with NetBoot shares not being browsable on macOS 10.14
* Updated to the latest version of Reposado
* Updated appliance build scripts for latest Operating Systems
* Fixed a security issue, relating to a LFI vulnerability from the Logs page
* Added macOS 10.15 SUS catalog options
* Added the ability to select the NetBoot engine (pybsdp/DHCP)
* Added (experimental) support for RHEL 8
* Defaults to Active Directory login, if it is configured
* Appliance Operating System is now Ubuntu 16.04.6
* Will preserve a Kinobi installation if detected during the Installation

## 5.0

* Updated user interface to align with Jamf Pro UI
* Appliance Operating System is now Ubuntu 16.04.5
* Added (experimental) support for Ubuntu 18.04 LTS
* Added notifications for recommended actions
* Improved Active Directory (LDAP) integration for web administration
* Added ability to create/manage local system accounts
* Added support for multiple network interfaces.\
  Note: Bonded interfaces are not supported in the UI
* Added ability to configure global proxy
* Improved ntp syncronization in Date/Time
* Added timezone picker map in Date/Time
* Improved SSL certificate interface
* Added full chain validation checks for SSL certificates
* New storage management interface for expanding LVM volumes
* Improved log viewer
* Added ability to download contents of log viewer
* Added dedicated settings pages for services
* Settings gear is contextually aware of service
* Added ability to enable/disable services
* Added ability to show/hide services in Dashboard
* Added ability to select SUS catalogs to sync
* Added ability to add custom SUS catalogs (beta/seed)
* Added ability to publish SUS catalogs over https
* Improved URL re-write functionality for SUS catalogs
* Added support for multiple NetBoot images
* Improved NetBoot image property editing
* Improved validation checks for LDAP proxy configuration
* Added service status messages to LDAP proxy
* Added File Sharing service
* Added ability to create smb/afp/http shares
* Added ability to manage users for shares

## 4.2.1

* Added High Sierra support for SUS
* Updated reposado to latest version
* Fixed an issue where certain Sierra builds would NetBoot extremely slowly

## 4.2

* Added Sierra support for SUS
* Updated reposado to latest version
* Added validation for SUS Base URL and Branch name(s) with live feedback
* Added (missing) option for SUS sync at 9:00 AM
* Improved detection of the last SUS sync date and time
* Added proxy configuration to SUS
* Added validation for NetBoot Image Name, Subnet and Netmask with live feedback
* Added checks for NetBoot supporting services
* Provisioned for NFS support for NetBoot Images
* Updated service controls for TFTP on RHEL/CentOS
* Added validation for Hostname, IP Address, Netmask, Gateway and DNS Servers with live feedback
* Added functionality to dynamically determine primary network interface, to allow for variations
* Updated network configuration to persistently configure static DNS on Ubuntu
* Updated service controls for SSH and Firewall
* Updated timezone configuration
* Added validation for Network Time Server with live feedback
* Added functionality to create CSR (and new Private Key) in webadmin GUI
* Added field descriptions for certificates
* Added functionality to view logs in the webadmin GUI
* Added Web UI for expanding the primary volume, when the underlying VMDK is expanded
* Added functionality to enable/disable AFP service
* Added functionality to enable/disable SMB service
* Updated about page to reflect OS and installed packages
* Updated Jamf Nation links
* Removed support for Ubuntu 10.04 and 12.04 (both are EOL)
* Added support for Ubuntu 16.04
* Removed hard-coded OS checks, replaced with detection of binaries or configuration files
* Improved detection/installation of supporting software
* Installer now updates existing files in-place, rather than overwriting with templates
* Updated mechanisms used for Ubuntu service controls to ensure services are correctly enabled / disabled
* Firewall rule configuration removed from adminHelper.sh, all firewall rules are pre-configured during installation
* Added 'enablegui' option to adminHelper.sh to easily re-enable webadmin GUI
* Fixed an issue where the LDAP Proxy prompts for a password during installation

## 4.1

* New and improved User Interface and minor changes to the User Experience
* Added LDAP administration group login support
* Added ability to rename the advertised NetBoot name
* Documentation updated, improved, and converted to markdown format

## 4.0

* Renamed to NetBoot/SUS/LP (NetSUSLP) for reference to LDAP Proxy
* Added El Capitan support for SUS
* Added firewall functionality with port managing for running NetSUSLP services by using app armor
* Added ability to disabled WebAdmin interface
* Added LDAP Proxy functionality with the use of slapd
* Added GAWK installation for WebAdmin on Ubuntu operating systems
* Added functionality to only enable services as needed
* Added functionality to update Ubuntu apt-get repository to prevent failures on service installation
* Added certificate page to allow tomcat or slapd certificates, and configured an installation to use a self-signed certificate
* Changed NetBoot page to enable SMB for uploading a NetBoot file, and then disable it when it is not in use

base/testOSRequirements.sh
#!/bin/bash

logNoNewLine "Checking for a supported OS..."

if [ -f "/etc/os-release" ]; then
	source /etc/os-release
elif [ -e "/etc/system-release" ]; then
	NAME=$(sed -e 's/ release.*//' /etc/system-release)
	PRETTY_NAME=$(sed -e 's/ release//' /etc/system-release)
	VERSION_ID=$(sed -e 's/.*release //;s/ .*//' /etc/system-release)
fi
if [[ -z "$NAME" ]]; then
	NAME=$(uname -s)
fi

case $NAME in
"Ubuntu")
	if [[ "$VERSION_ID" == "14.04" ]] || [[ "$VERSION_ID" == "16.04" ]] || [[ "$VERSION_ID" == "18.04" ]] ; then
		log "$PRETTY_NAME found"
		if [[ "$VERSION_ID" == "18.04" ]] ; then
			log "Warning: $NAME $VERSION_ID support is currently experimental, proceed with caution."
		fi
		exit 0
	else
		log "Error: $NAME version must be 14.04, 16.04 or 18.04 (Detected $VERSION_ID)."
		exit 1
	fi
;;
"Red Hat Enterprise Linux"*|"CentOS"*)
	IFS='.'
	VERSION_ARR=( $VERSION_ID )
	unset IFS
	if [[ ${VERSION_ARR[0]} -eq 6 ]] && [[ ${VERSION_ARR[1]} -gt 3 ]] || [[ ${VERSION_ARR[0]} -gt 6 ]] ; then
		log "$PRETTY_NAME found"
		if [[ ${VERSION_ARR[0]} -eq 8 ]] ; then
			log "Warning: $NAME $VERSION_ID support is currently experimental, proceed with caution."
		fi
		exit 0
	else
		log "Error: $NAME version must be 6.4 or later (Detected $VERSION_ID)."
		exit 1
	fi
	if yum repolist | grep repolist | grep -q ': 0'; then
        log "Error: This system is does not have any available repositories."
		exit 1
    fi
;;
*)
	release=$(rpm -q --queryformat '%{RELEASE}' rpm | cut -d '.' -f 2)
	if [[ $release == "el6" ]] || [[ $release == "el7" ]] || [[ $release == "el8" ]] ; then
		IFS='.'
		VERSION_ARR=( $VERSION_ID )
		unset IFS
		if [[ ${VERSION_ARR[0]} -eq 6 ]] && [[ ${VERSION_ARR[1]} -gt 3 ]] || [[ ${VERSION_ARR[0]} -gt 6 ]] ; then
			log "$PRETTY_NAME found"
			log "Warning: $NAME is a Red Hat Enterprise Linux variant, proceed with caution."
			exit 0
		else
			log "Error: $NAME version must be 6.4 or later (Detected $VERSION_ID)."
			exit 1
		fi
	fi
	log "Error: Did not detect a valid Ubuntu/Red Hat/CentOS install (Detected $NAME)."
	exit 1
;;
esac
